conn.go

```diff
@@ -222,6 +219,20 @@ func (c *Conn) authAllowed() bool {
 	return !c.server.AuthDisabled && (isTLS || c.server.AllowInsecureAuth)
 }
 
+// protocolError writes errors responses and closes the connection once too many
+// have occurred.
+func (c *Conn) protocolError(code int, ec EnhancedCode, msg string) {
+	c.WriteResponse(code, ec, msg)
+
+	c.errCount++
+	if c.errCount > errThreshold {
+		// 421 -> Connection closing (RFC5321 Section-4.2.3)
+		// 5.7.0 -> Persistent security related issue
+		c.WriteResponse(421, EnhancedCode{5, 7, 0}, "Too many errors. Quiting now")
```

Comment:
Sorry for that back-and-forth with status codes but as I mentioned [earlier](https://github.com/emersion/go-smtp/pull/116#discussion_r475574014), 500 5.5.X (0?) is probably what should be used in this particular case. 

Otherwise, LGTM.

Reply:
Np. I'll switch that now :D